dlib/dnn/trainer_abstract.h

```diff
@@ -96,6 +99,14 @@ namespace dlib
                       cudaGetDevice()).  In addition, you can ask to use additional
                       devices, which you do by putting their device numbers into
                       cuda_extra_devices.
+                - if (thread_pools.get() != nullptr) then
+                    - Any new threads spun within the trainer will execute within the
+                      passed thread pools vector. This means that the same threads can
+                      be re-used across different dnn_trainer instances. Otherwise, the
+                      CUDA runtime may leak memory. This, however, is relevant only if
+                      your program is going to instantiate a large number of trainers,
+                      and generally stay up and running for a very long time. If not,
+                      then you need not worry about this.
```

Comment:
Make an extra note that the CUDA runtime memory leak isn't because dlib code is not calling appropriate cuda free calls, but is a limitation of the CUDA runtime we have no control over.

Reply:
Also note that any particular thread pool vector should only be passed to one trainer instance at a time.

Reply:
Added these notes.